biomass/GEOCARBON/convert.py

```diff
@@ -0,0 +1,99 @@
+import os
+import time
+
+import cftime
+import numpy as np
+import xarray as xr
+
+
+def CoarsenDataset(filename, res=0.5, ntile=None):
+    """Coarsens the source dataset by using xarray and dask to avoid large memory.
+
+    Parameters
+    ----------
+    filename : str
+        the netcdf file to coarsen
+    outfile : str
+        the name of the file for the coarsened output
+    res : float, optional
+        the approximate resolution of the coarsen output in degrees
+    ntile : int, optional the number of coarse cells to process at a
+        time, routine will process a ntile x ntile subgrid. Increase
+        to run faster, decrease to reduce the peak memory.
+
+    """
+    with xr.open_dataset(filename) as ds:
+        ds = ds.rename({"latitude": "lat", "longitude": "lon"})
+        res0 = float(ds.lon.diff(dim="lon").mean())
+        n = int(round(res / res0))
+        if ntile is not None:
+            ds = ds.chunk({"lat": ntile * n, "lon": ntile * n})
+        c = ds.coarsen({"lat": n, "lon": n}, boundary="pad").mean()
+    return c
+
+
+if __name__ == "__main__":
+
+    # If local file does not exist, download
+    remote_source = "https://www.bgc-jena.mpg.de/geodb/projects/DataDnld.php"
```

Comment:
This link requires some login information from the user. Maybe note that or use a different link?

Reply:
Also wondering if we should embed a way to download within the script. I think I did that in my GFW convert.py but I might not have for HWSD2.

Reply:
I would love to be able to directly download but unfortunately this data provider does not allow that. Once you request a dataset they create a link to the file to send to you that also contains the date. You have a few days to use that link because they purge them regularly. I think this practice is silly in today's data availability climate but some sites are slow to adapt to new trends or go back and update old data.

Reply:
Yeah, I really don't understand guarding data unless people just don't have the resources to store it on a server. 

The link you have leads me to a login page, so maybe you could send folks to one directory up?https://www.bgc-jena.mpg.de/geodb/projects/FileDetails.php
There's also this link I found to request the data via email: https://www.wur.nl/en/research-results/chair-groups/environmental-sciences/laboratory-of-geo-information-science-and-remote-sensing/research/integrated-land-monitoring/forest_biomass/forest-biomass-downloads.htm

Either I think are better than the login page!

Reply:
I am sure I had the project home page up, but when you copy paste that link, it doesn't take you there. This portal is terrible, we also use FLuxcom stuff which is worse. 👎🏻